### PR TITLE
Code review comments for https://github.com/omhpe/oauth2-proxy/pull/1

### PR DESCRIPTION
**Pull request by omhpe**
The code review comments done for **https://github.com/omhpe/oauth2-proxy/pull/1** are not visible since the branch mistakenly got merged. 

This PR has code review comments addressed for **https://github.com/omhpe/oauth2-proxy/pull/1**

### CODE REVIEW

File: pkg/sessions/redis/redis_store.go [hunk 1]
```diff
@@ -103,6 +101,20 @@ func (store *SessionStore) ClearAll(ctx context.Context, s *sessions.SessionStat
 	return nil
 }
 
+func saveTicketWithLock(ctx context.Context, store *SessionStore, keyName string, ticket string, exp time.Duration) error {
+	lock := store.Lock(ticket)
```
Comment (joel-bluedata) [inline pkg/sessions/redis/redis_store.go]: Seems like we should still be using keyName for the lock argument here?

File: pkg/sessions/redis/redis_store.go [hunk 2]
```diff
@@ -114,20 +126,33 @@ func (store *SessionStore) SaveUserSession(ctx context.Context, s *sessions.Sess
...
 		ticket = string(keyVal)
 	}
-	err := s.ObtainLock(ctx, exp)
-	if err != nil {
-		return fmt.Errorf("error occurred while trying to obtain lock: %v", err)
-	}
-	defer func() {
-		if s == nil {
-			return
+	return saveTicketWithLock(ctx, store, keyName, ticket, exp)
```
Comment (joel-bluedata) [inline pkg/sessions/redis/redis_store.go]: So... this function is generally speaking doing a "read-modify-write" of the document stored under keyName. As the code currently stands, it doesn't get the lock until it is about to do the write. Instead, the lock needs to be obtained before the read (to be useful in avoiding races).

File: pkg/sessions/redis/redis_store.go [hunk 3]
```diff
@@ -114,20 +126,33 @@ func (store *SessionStore) SaveUserSession(ctx context.Context, s *sessions.Sess
...
+}
+
+func (store *SessionStore) ClearUserSession(ctx context.Context, s *sessions.SessionState, value string, exp time.Duration) error {
```
Comment (joel-bluedata) [inline pkg/sessions/redis/redis_store.go:133]: Similar comment about the locking in this clear function... the lock needs to be obtained before the "read" in the "read-modify-write".

**Comment by joel-bluedata** [comment]
Generally looks good, just some comments about locking.

File: pkg/sessions/redis/redis_store.go (hunk 1 above)
Comment (omhpe) [reply pkg/sessions/redis/redis_store.go]: yes, done the change

File: pkg/sessions/redis/redis_store.go (hunk 2 above)
Comment (omhpe) [reply pkg/sessions/redis/redis_store.go]: thanks @joel-bluedata . done the change

File: pkg/sessions/redis/redis_store.go (hunk 3 above)
Comment (omhpe) [reply pkg/sessions/redis/redis_store.go:133]: done @joel-bluedata . thanks

**Review by joel-bluedata — APPROVED** [review]

**Pull request merged**